src/entity/phases/types.ts

```diff
@@ -0,0 +1,48 @@
+import type { TasksUrls } from '../../types/config.type'
+
+export type PhaseDescriptor = Readonly<{
+  number: number
+  name: string
+  startDate: string
+  endDate: string
+}>
+
+export type Duration = Readonly<{
+  from: string
+  to: string
+}>
+
+export type Task = Readonly<{
+  summary: string
+  description?: JSX.Element
+  rewards?: (points: number) => JSX.Element
+  judgingCriteria?: JSX.Element
+  submit?: (urls: TasksUrls) => JSX.Element
+  duration?: Duration
+  completed?: boolean
+  points: number
+}>
```

Comment:
I'm quite embarrassed with the changes you made in the `entity` files, because the declared object types are not entities in terms on business model, there are more a view model that brings to the view some extra properties for a dumb usage.

An entity is a front-end object (the model) that allows to persist you business properties that could be then computed to the view.

It's a quite important issue because this type will be shared between multiple components and does not represent the model itself.

As an example, an entity does know what is a `JSX.Element` ...